### PR TITLE
ENH: m1nn searchlight can now do correlation distance

### PR DESCRIPTION
**Pull request by yarikoptic**
As demanded by Sam, so for him to try/review/fix/improve

### CODE REVIEW

File: mvpa2/measures/nnsearchlight.py [hunk 1]
```diff
@@ -44,7 +44,15 @@ def __init__(self, knn, generator, qe, **kwargs):
           Used to fetch space and dfx settings. TODO
         """
         # verify that desired features are supported
-        if knn.dfx != squared_euclidean_distance:
+        if knn.dfx == squared_euclidean_distance:
+            self._distance = 'euclidean'
+        elif knn.dfx == one_minus_correlation:
+            self._distance = 'correlation'
```
Comment (yarikoptic) [inline mvpa2/measures/nnsearchlight.py]: and coveralls claims that this code is not tested, not sure how that could have happened

**Comment by yarikoptic** [comment]
@snastase   so it worked without puking?  as we discussed it could be made more efficient through alternative way to compute it but not sure if I am to code it atm, so would like to merge

**Comment by snastase** [comment]
@yarikoptic yes, works without puking and verified that it produces the same result as more generic (slow) kNN searchlight using one_minus_correlation dfx on a real fMRI dataset (up to a couple divide-by-zero errors)

**Comment by yarikoptic** [comment]
good, merging then

**Pull request merged**